### ISSUE TITLE
Flutter FCM for iOS - Reports Issue

### ISSUE THREAD

**Issue by karthikeyanp307** [FirebaseExtended/flutterfire#5529]
Hi,

I am using firebase_messaging  v9.0.1 in my Flutter application. On configuring this library based on the https://pub.dev/packages/firebase_messaging/example, I am able to receive the notification for android in both foreground and background states. But the same for iOS is not working. I am getting the below error,

APNS device token not set before retrieving FCM Token for Sender ID ''. Notifications to this FCM Token will not be delivered over APNS.Be sure to re-retrieve the FCM token once the APNS device token is set.

Do I want to call any other function apart from FirebaseMessaging.instance.getToken() for iOS? Please help.

**Comment by darshankawar** [comment]
@karthikeyanp307 
Can you check and confirm if your iOS device is connected properly to the internet ?
If so, does it return token from `FirebaseMessaging.instance.getToken()` ?

Also, provide your `flutter doctor -v` to properly address this issue.
Check if any of these helps.

https://stackoverflow.com/a/66123064

**Comment by sguru049** [comment]
@darshankawar I'm also facing the same issue for my web-app, with firebase_messaging v9.0.1. The notifications doesn't work in iOS devices, for any browsers. For Android, It works good for Chrome but fails for Firefox and Opera.

`FirebaseMessaging.instance.getToken()` returns `null` for all browsers on iOS. Same for Firefox and Opera on Android. The issue is, the notification asking for permission to send notifications doesn't pop-up on these devices. 

`flutter doctor -v` also shows "No issues Found!"

Can you please suggest a solution ?

**Comment by peteroleeman** [comment]
having same issue as well

**Comment by karthikeyanp307** [comment]
@darshankawar  - I tried with the https://github.com/FirebaseExtended/flutterfire/tree/master/packages/firebase_messaging/firebase_messaging/example. When configured with FCM - google-services it works fine for android and getting the same error for iOS. 

APNS device token not set before retrieving FCM Token for Sender ID ''. Notifications to this FCM Token will not be delivered over APNS.Be sure to re-retrieve the FCM token once the APNS device token is set.My iOS device is connected to internet and there were no network related issues while running this.

Also getAPNSToken() always returns null. Please help me on this. 

Thanks.

**Comment by darshankawar** [comment]
@karthikeyanp307 
It seems you haven't setup your iOS part properly. See this document and follow the steps as directed, https://firebase.flutter.dev/docs/messaging/apple-integration.

Also, the document states that, `On iOS & macOS, you must have correctly setup your project to integrate with FCM and APNs.`

**Comment by karthikeyanp307** [comment]
@darshankawar - Thanks for your reply.  

I used firebase_messaging: ^7.0.3 earlier with the same application and both android & iOS were working fine. When upgraded the same application to the latest firebase messaging v9.0.1 I started getting this error for iOS.But the same works fine for android. Is there anything specific that we need to do, while upgrading to the latest firebase messaging library. 

Thanks.
